### PR TITLE
Remove usages of `pushd`/`popd`

### PR DESCRIPTION
**Pull request by erichaagdev**
Note: I updated CI to run the same set of tests as the cross platform tests. You can see a successful execution of that here: https://github.com/gradle/gradle-enterprise-build-validation-scripts/actions/runs/3680255149/jobs/6225649761

### CODE REVIEW

File: .github/workflows/build-verification.yml [hunk 1]
```diff
@@ -25,3 +25,69 @@ jobs:
         run: ./gradlew build -i
         env:
           GRADLE_ENTERPRISE_ACCESS_KEY: ${{ secrets.GE_SOLUTIONS_ACCESS_TOKEN }}
+      - name: Download and extract build validation Scripts
```
Comment (jthurne) [inline .github/workflows/build-verification.yml:28]: Are these additions intentional?

Comment (erichaagdev) [reply .github/workflows/build-verification.yml]: Indeed. I feel like we're missing simple tests like this for our PR branches. Though the name of this step is incorrect and has been fixed.

Let me know if you think I should remove these tests.

Comment (jthurne) [reply .github/workflows/build-verification.yml]: I don't have any objection to them, so long as they don't take a crazy amount of time to run (like a an hour or two).

Comment (erichaagdev) [reply .github/workflows/build-verification.yml]: The run for this PR took only 2 minutes. 😄 

https://github.com/gradle/gradle-enterprise-build-validation-scripts/actions/runs/3680308034/jobs/6225766497

**Review by jprinet — APPROVED** [review]
LGTM 💪
Could you try the [AndroidX workflow](https://github.com/gradle/gradle-enterprise-oss-projects/actions/runs/3668257911/workflow) to make sure everything's fine when project_dir has to be set?

File: .github/workflows/build-verification.yml [hunk 2]
```diff
@@ -25,3 +25,69 @@ jobs:
         run: ./gradlew build -i
         env:
           GRADLE_ENTERPRISE_ACCESS_KEY: ${{ secrets.GE_SOLUTIONS_ACCESS_TOKEN }}
+      - name: Extract build validation Scripts
```
Comment (etiennestuder) [inline .github/workflows/build-verification.yml:28]: I would make the new stuff here at least a separate job. Possibly, run sequentially after the `Verification` job.

Comment (etiennestuder) [inline .github/workflows/build-verification.yml:28]: We should add the same verification steps to the `development-release.yml` workflow which runs on master.

**Review by etiennestuder — CHANGES_REQUESTED** [review]

File: .github/workflows/build-verification.yml (hunk 2 above)
Comment (erichaagdev) [reply .github/workflows/build-verification.yml]: I will address CI in a separate PR. I have removed these changes for now.

**Review by etiennestuder — APPROVED** [review]
I did not actually run the scripts with a version that includes this PR. I assume @erichaagdev does/did that. Code LGTM.

**Comment by etiennestuder** [comment]
> LGTM 💪 Could you try the [AndroidX workflow](https://github.com/gradle/gradle-enterprise-oss-projects/actions/runs/3668257911/workflow) to make sure everything's fine when project_dir has to be set?

@erichaagdev that seems like a good idea.

**Comment by erichaagdev** [comment]
**_WIP_**

## Testing

### `exp1-gradle`

```shell
./01-validate-incremental-building.sh \
  -s "https://ge.solutions-team.gradle.com" \
  -r "https://github.com/gradle/androidx" \
  -t "buildOnServer zipTestConfigsWithApks test" \
  -p "biometric" \
  -a "-x ktlint"
```

### `exp2-gradle`

```shell
./02-validate-local-build-caching-same-location.sh \
  -s "https://ge.solutions-team.gradle.com" \
  -r "https://github.com/gradle/androidx" \
  -t "buildOnServer zipTestConfigsWithApks test" \
  -p "biometric" \
  -a "-x ktlint"
```

### `exp3-gradle`

```shell
./03-validate-local-build-caching-different-locations.sh \
  -s "https://ge.solutions-team.gradle.com" \
  -r "https://github.com/gradle/androidx" \
  -t "buildOnServer zipTestConfigsWithApks test" \
  -p "biometric" \
  -a "-x ktlint"
```

**Comment by erichaagdev** [comment]
@etiennestuder @jprinet please see the tests I ran above. ☝️ 

I will merge this PR given the successful outcome.

**Pull request merged**